homeassistant/components/roborock/sensor.py

```diff
@@ -0,0 +1,118 @@
+"""Support for Roborock sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from roborock.typing import DeviceProp
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.util import slugify
+
+from .const import DOMAIN
+from .coordinator import RoborockDataUpdateCoordinator
+from .device import RoborockCoordinatedEntity
+from .models import RoborockHassDeviceInfo
+
+
+@dataclass
+class RoborockSensorDescriptionMixin:
+    """A class that describes sensor entities."""
+
+    value_fn: Callable[[DeviceProp], int]
+
+
+@dataclass
+class RoborockSensorDescription(
+    SensorEntityDescription, RoborockSensorDescriptionMixin
+):
+    """A class that describes Roborock sensors."""
+
+
+CONSUMABLE_SENSORS = [
+    RoborockSensorDescription(
```

Comment:
These seem like the should be diagnostic sensors. I checked xiaomi and it also uses these as diagnostics.

Reply:
Makes sense!